src/libmongoc/tests/unified/result.c

```diff
@@ -349,11 +349,6 @@ result_check (result_t *result,
 
    /* check result. */
    if (expect_result) {
-      if (!result->ok) {
-         test_set_error (
-            error, "expected result, but got error: %s", result->error.message);
-         goto done;
-      }
```

Comment:
This was an existing bug in the unified test runner. We should not necessarily be expecting the result to be ok when `expectEvents` are specified according to the [spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectedevent). I'll handle this change through a separate PR and ticket and rebase this PR eventually.